### PR TITLE
fix: filter should split by brackets or symbols

### LINKED ISSUE #119: Bug: 关键词为中文时关键词被过滤掉了
### Describe the bug

我加载了很多别人生成好的 Textual Inversion embeddings ，有些 embedding 的关键词是中文，输入的时候被过滤掉了。关键词为英文的 embedding 则正常。

### Steps to reproduce

[embeddings.zip](https://github.com/koishijs/novelai-bot/files/10047142/embeddings.zip)
这是五个embedding文件，op-ganyu、op-naxida关键词为英文，申鹤、雏鹤爱、霞泽美游的关键词为中文。

首先我关闭了自动翻译和只接受英文输入。卸载了所有翻译插件。

违禁词列表只有nude, naked。

直接使用webui时输入`申鹤`这个关键词AI是认识的：

但是使用bot时，给出的却是不相干的人物。使用png info查看后发现`申鹤`这个关键词就没有输入进去：

### Expected behavior

如果使用英文关键词`op-ganyu`的话，能正常显示人物图片，并且png info里也有输入的英文关键词：

### Screenshots

_No response_

### Relevant log output

_No response_

### Launcher

Koishi CLI (koishi start)

### Backend

Stable Diffusion WebUI (AUTOMATIC1111)

### Versions

- OS: Windows 11
- Adapter: onebot
- Node version: 16.18.0
- Koishi version: 4.10.2

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by MaikoTan**
fix #119

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -278,7 +278,7 @@ export function parseInput(input: string, config: Config, forbidden: Forbidden[]
 
   // remove forbidden words
   const positive = input.split(/,\s*/g).filter((word) => {
-    word = word.replace(/[^a-z0-9]+/g, ' ').trim()
+    word = word.replace(/[.:_<>|\\\(\)\[\]]+/g, ' ').trim()
```
Comment (shigma) [inline src/config.ts:281]: also need to add `\s` and other punctuations (quotation marks, etc.)

Comment (shigma) [reply src/config.ts:281]: or simply exclude all of ascii characters except a-z 0-9

**Review by shigma — CHANGES_REQUESTED** [review]

File: src/config.ts (hunk 1 above)
Comment (MaikoTan) [reply src/config.ts]: what about this?

File: src/config.ts [hunk 2]
```diff
@@ -278,7 +278,7 @@ export function parseInput(input: string, config: Config, forbidden: Forbidden[]
 
   // remove forbidden words
   const positive = input.split(/,\s*/g).filter((word) => {
-    word = word.replace(/[^a-z0-9]+/g, ' ').trim()
+    word = word.replace(/[\x00-\x7f]/g, (s) => s.replace(/[^0-9a-zA-Z]/, ' ')).trim() // ascii except /a-zA-Z0-9/
```
Comment (shigma) [inline src/config.ts:281]:
```suggestion
    word = word.replace(/[\x00-\x7f]/g, s => s.replace(/[^0-9a-zA-Z]/, ' ')).replace(/\s+/, ' ').trim()
```

should also handle consecutive spaces

Comment (MaikoTan) [reply src/config.ts]: didn't know the neccessariness, but ok

**Review by shigma — APPROVED** [review]

**Pull request merged**